course_info/tests/jasmine/controllers/DetailsControllerSpec.js

```diff
@@ -559,4 +566,161 @@ describe('Unit testing DetailsController', function () {
+            expect(dc.courseInstance.sites.length).toEqual(3);
+            expect(dc.courseInstance.sites).toEqual(siteList);
+        });
+
+        it('should show an error message if the post fails', function(){
+            dc.newCourseSiteURL = 'http://testing.com';
+            dc.associateNewSite();
+            $httpBackend.expectPOST(postSiteUrl)
+                .respond(500, JSON.stringify({status: "invalid url"}));
+            $httpBackend.flush(1);
+            expect(dc.alerts.form.siteOperationFailed).toEqual({show: true, operation: 'associating', details: 'None'});
+        });
+
+    });
+
+    describe('dissociateSite', function(){
+        beforeEach(function () {
+            var ci = {
+                course_instance_id: $routeParams.courseInstanceId,
+                title: 'Test Course Title',
+                short_title: 'Test',
+                sub_title: 'Jasime is your friend',
+                description: '<p>hello</p>',
+                sync_to_canvas : 1,
+                exclude_from_isites:0,
+                exclude_from_catalog:0,
+                sites: [
+                    {
+                        external_id: 'https://x.y.z/888',
+                        site_id: '888',
+                        map_type: 'official',
+                        course_site_url: 'https://x.y.z/888',
+                        site_map_id: '888',
+                    },
+                    {
+                        external_id: 'https://x.y.z/999',
+                        site_id: '999',
+                        map_type: 'unofficial',
+                        course_site_url: 'https://x.y.z/999',
+                        site_map_id: '999',
+
+                    }
+                ]
+            };
+
+            dc = $controller('DetailsController', {$scope: scope});
+            dc.courseInstance = ci;
+            $httpBackend.expectGET(courseInstanceURL).respond(200, JSON.stringify({status: "success"}));
+            $httpBackend.expectGET(peopleURL).respond(200, JSON.stringify({status: "success"}));
+            $httpBackend.flush(2);
+        });
+
+        it('should make the delete call to disasscociate a url with the course instance', function(){
+            //delete by specifying the index
+            siteListIndex = 1;
+            var siteURL = '';
+            var site_map_id = '';
+
+            dc.dissociateSite(siteListIndex);
+            scope.$digest();// resolves modal
+            dc.confirmDissociateSiteModalInstance.close(siteURL, site_map_id);
+            $httpBackend.expectDELETE(deleteSiteUrl).respond(204, {});
+            $httpBackend.flush(1);
+            expect(dc.courseInstance.sites.length).toEqual(dc.courseInstance.sites.length);
+        });
+
+        it('doesn\'t delete any sites if the process if canceled', function() {
+            dc.dissociateSite(siteListIndex);
+            scope.$digest()
+            dc.confirmDissociateSiteModalInstance.dismiss()
```

Comment:
Have initialized the index, not sure why it din't throw an error before. And validating no change in count and also added a isUndefined check